linux/tests/socket_transport_linux_ut/socket_transport_linux_ut.c

```diff
@@ -0,0 +1,1485 @@
+    g_test_socket_array[test_socket++] = real_gballoc_ll_malloc(sizeof(SOCKET_HANDLE));
+    return test_socket-1;
+}
+
+static int my_close(int s)
+{
+    test_socket--;
+    real_gballoc_ll_free((void*)g_test_socket_array[s]);
+    return 0;
+}
+
+static const char* my_inet_ntop(int af, const void* cp, char* buf, socklen_t len)
+{
+    (void)af;
+    (void)cp;
+    (void)len;
+
+    strcpy(buf, TEST_INCOMING_HOSTNAME);
+    return buf;
+}
+
+static int my_getaddrinfo(const char* pNodeName, const char* pServiceName, const struct addrinfo* pHints, struct addrinfo** ppResult)
+{
+    (void)pNodeName;
+    (void)pServiceName;
+    (void)pHints;
+    *ppResult = &g_addrInfo;
+    return 0;
+}
+
+static ssize_t my_send(SOCKET_HANDLE sockfd, const void* buf, size_t len, int flags)
+{
+    (void)sockfd;
+    (void)buf;
+    (void)flags;
+    return len;
+}
+
+static ssize_t my_recv(SOCKET_HANDLE sockfd, void* buf, size_t len, int flags)
+{
+    (void)sockfd;
+    (void)flags;
+
+    memcpy(buf, recv_buff, len);
+    return TEST_BYTES_RECV;
+}
+
+BEGIN_TEST_SUITE(TEST_SUITE_NAME_FROM_CMAKE)
+
+TEST_SUITE_INITIALIZE(suite_init)
+{
+    ASSERT_ARE_EQUAL(int, 0, umock_c_init(on_umock_c_error), "umock_c_init failed");
+    ASSERT_ARE_EQUAL(int, 0, umocktypes_stdint_register_types(), "umocktypes_stdint_register_types failed");
+    ASSERT_ARE_EQUAL(int, 0, umocktypes_bool_register_types(), "umocktypes_bool_register_types failed");
+
+    REGISTER_SM_GLOBAL_MOCK_HOOK();
+
+    REGISTER_GLOBAL_MOCK_HOOK(malloc, real_gballoc_ll_malloc);
+    REGISTER_GLOBAL_MOCK_FAIL_RETURN(malloc, NULL);
+    REGISTER_GLOBAL_MOCK_HOOK(free, real_gballoc_ll_free);
+
+    REGISTER_GLOBAL_MOCK_HOOK(socket, my_socket);
+    REGISTER_GLOBAL_MOCK_FAIL_RETURN(socket, INVALID_SOCKET);
+    REGISTER_GLOBAL_MOCK_HOOK(close, my_close);
+    REGISTER_GLOBAL_MOCK_RETURN(bind, 0);
+    REGISTER_GLOBAL_MOCK_FAIL_RETURN(bind, -1);
+    REGISTER_GLOBAL_MOCK_RETURN(listen, 0);
+    REGISTER_GLOBAL_MOCK_FAIL_RETURN(listen, -1);
+    REGISTER_GLOBAL_MOCK_HOOK(accept, my_accept);
+    REGISTER_GLOBAL_MOCK_FAIL_RETURN(accept, INVALID_SOCKET);
+    REGISTER_GLOBAL_MOCK_HOOK(inet_ntop, my_inet_ntop);
+    REGISTER_GLOBAL_MOCK_HOOK(getaddrinfo, my_getaddrinfo);
+    REGISTER_GLOBAL_MOCK_FAIL_RETURN(getaddrinfo, MU_FAILURE);
+    REGISTER_GLOBAL_MOCK_FAIL_RETURN(connect, MU_FAILURE);
+    REGISTER_GLOBAL_MOCK_HOOK(send, my_send);
+    REGISTER_GLOBAL_MOCK_FAIL_RETURN(send, INVALID_SOCKET);
+    REGISTER_GLOBAL_MOCK_HOOK(recv, my_recv);
+    REGISTER_GLOBAL_MOCK_FAIL_RETURN(recv, INVALID_SOCKET);
+
+    REGISTER_TYPE(SM_RESULT, SM_RESULT);
+
+    REGISTER_UMOCK_ALIAS_TYPE(SM_HANDLE, void*);
+    REGISTER_UMOCK_ALIAS_TYPE(SOCKET_HANDLE, int);
+    REGISTER_UMOCK_ALIAS_TYPE(ssize_t, long);
+    REGISTER_UMOCK_ALIAS_TYPE(socklen_t, int);
+
+    g_addrInfo.ai_canonname = "ai_canonname";
+    for (size_t index = 0; index < TEST_BYTES_RECV; index++)
+    {
+        recv_buff[index] = 'a';
```

Comment:
Consider initializing with something more aggressive for testing - like 'a', 'b', ...